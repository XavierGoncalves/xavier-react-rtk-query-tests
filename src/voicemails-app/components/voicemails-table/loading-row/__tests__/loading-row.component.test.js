import React from 'react'
import { render } from '@testing-library/react'
import { LoadingRow } from '..'

describe('<LoadingRow />', () => {
  describe('when small prop is false', () => {
    it('renders the large loading row', () => {
      const { container } = renderComponent(false)

      expect(container).toMatchSnapshot()
    })
  })

  describe('when small prop is true', () => {
    it('renders the small loading row', () => {
      const { container } = renderComponent(true)

      expect(container).toMatchSnapshot()
    })
  })
})

const renderComponent = small =>
  render(
    <table>
      <tbody>
        <LoadingRow small={small} />
      </tbody>
    </table>
  )
