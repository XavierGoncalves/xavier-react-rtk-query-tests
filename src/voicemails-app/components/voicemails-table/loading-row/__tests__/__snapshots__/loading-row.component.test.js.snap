// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingRow /> when small prop is false renders the large loading row 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="co-table__row"
      >
        <td
          class="co-table__cell"
        >
          <div
            class="co-placeholder co-placeholder--animated co--medium"
          />
        </td>
        <td
          class="co-table__cell"
        >
          <div
            class="co-placeholder co-placeholder--animated co--small"
          />
        </td>
        <td
          class="co-table__cell"
        >
          <div
            class="co-placeholder co-placeholder--animated co--medium"
          />
        </td>
        <td
          class="co-table__cell"
        >
          <div
            class="co-placeholder co-placeholder--animated co--medium"
          />
        </td>
        <td
          class="co-table__cell"
        >
          <div
            class="co-placeholder co-placeholder--animated co--small"
          />
        </td>
        <td
          class="co-table__cell"
        >
          <div
            class="co-placeholder co-placeholder--animated co--medium"
          />
        </td>
        <td
          class="co-table__cell"
        >
          <div
            class="co-placeholder co-placeholder--animated co--small"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<LoadingRow /> when small prop is true renders the small loading row 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="co-table__row"
      >
        <td
          class="co-table__cell"
        >
          <div
            class="co-placeholder co-placeholder--animated co--large"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
